android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.java

```diff
@@ -161,6 +166,9 @@ public final void handle(MotionEvent event) {
       }
       return;
     }
+    if (mNumberOfPointers != event.getPointerCount()) {
+      mNumberOfPointers = event.getPointerCount();
```

Comment:
You may want to move it up a bit as there is some logic that may cause the gesture to fail in which case it would emit an event with an incorrect number of pointers